src/models/Message.js

```diff
@@ -0,0 +1,50 @@
+import User from "./User";
+
+class Message {
+  static mimicRealmMappedPropertiesSchema( msg ) {
+
+    const fromUser = User.mapApiToRealm( msg.from_user );
+    const toUser = User.mapApiToRealm( msg.to_user );
+
+    return {
+      ...msg,
+      subject: msg.subject,
+      body: msg.body,
+      fromUser,
+      toUser,
+      threadId: msg.thread_id
+    };
+  }
+
+  static createMessageForRealm( msg, realm ) {
+    const newMsg = {
+      ...msg,
+        subject: msg.subject,
+        body: msg.body,
+        fromUser: User.mapApiToRealm( msg.from_user, realm ),
+        toUser: User.mapApiToRealm( msg.to_user, realm ),
+        threadId: msg.thread_id
+    };
+
+    // need to append user after the observation realm object has been created
+    //delete newObs.user; //Q: Why do we need to do this?
```

Comment:
We don't, actually... this was code I wrote before I realized we can upsert data into Realm at the time of creation. This is fixed in the branch I'm working on (30-sound-recorder) and will likely get merged in next week.